src/pandablocks_ioc/_hdf_ioc.py

```diff
@@ -523,6 +549,74 @@ def _parameter_validate(self, record: RecordWrapper, new_val) -> bool:
             return False
         return True
 
+    async def _update_directory_path(self, new_val) -> None:
+        """Handles writes to the directory path PV, creating
+        directories based on the setting of the CreateDirectory record"""
+        new_path = Path(new_val).absolute()
+        create_dir_depth = self._create_directory_record.get()
+        max_dirs_to_create = 0
+        if create_dir_depth < 0:
+            max_dirs_to_create = abs(create_dir_depth)
+        elif create_dir_depth > len(new_path.parents):
+            max_dirs_to_create = 0
+        elif create_dir_depth > 0:
+            max_dirs_to_create = len(new_path.parents) - create_dir_depth
+
+        logging.debug(f"Permitted to create up to {max_dirs_to_create} dirs.")
+        dirs_to_create = 0
+        for p in reversed(new_path.parents):
+            if not p.exists():
+                if dirs_to_create == 0:
+                    # First directory level that does not exist, log it.
+                    logging.error(f"All dir from {str(p)} and below do not exist!")
+                dirs_to_create += 1
+            else:
+                logging.info(f"{str(p)} exists")
+
+        # Account for target path itself not existing
+        if not os.path.exists(new_path):
+            dirs_to_create += 1
+
+        logging.debug(f"Need to create {dirs_to_create} directories.")
+
+        # Default message is "OK"
+        status_msg = "OK"
+
+        # Case where all dirs exist
+        if dirs_to_create == 0:
+            if os.access(new_path, os.W_OK):
+                self._directory_exists_record.set(1)
+            else:
+                status_msg = "Dirs exist but aren't writable."
+                self._directory_exists_record.set(0)
+        # Case where we will create directories
+        elif dirs_to_create <= max_dirs_to_create:
+            logging.info(f"Attempting to create {dirs_to_create} dir(s)...")
```

Comment:
Either this should be debug like the next log on success, or both should be info.